specs/collect_brick_device_details.adoc

```diff
@@ -0,0 +1,116 @@
+= Collect Brick device details
+
+The underlying device details of gluster bricks has to be maintained in tendrl, as it helps
+admins to take corrective actions on the devices when the brick goes bad.
+
+
+== Problem description
+
+Gluster is just aware of the mount point that is used to create the bricks, it does not worry
+about the underlying devices as it operates with mountpoints. But if something goes wrong with
+the brick, its not of much help if tendrl reports that there is problem with the mount point
+as it is a logical entity and the root cause for this would be an underlying device. So tendrl
+should maintain the mapping between the bricks and underlying devices.
+
+
+== Use Cases
+
+* It helps the storage admin to easily take corrective actions when some thing happens to the
+Brick.
+
+== Proposed change
+
+Since brick device details are not provided by gluster get state. Tendrl should explicitly
+gather these details by using the mount point of the brick.
+
+For gathering this information tendrl can use blivet python module. Blivet is a python module
+for system storage configuration. Using blivet tendrl can gather the device hierarcy that is
+used to create the brick. Sample snippet of how to gather the information from blivet is given
+below:
+[source,python]
+----
+import blivet
+
+b = blivet.Blivet()
+b.reset() # this will gather the device information in this machine
+device = b.devicetree.resolveDevice(<filesystem>)
+
+ancestors = device.ancestors
+----
+
+=== Alternatives
+
+None
+
+=== Data model impact
+
+Data model remains the same, these details will be stored in the brick object, whose
```

Comment:
These were already there but. It was updated only when we provisioned bricks through tendrl. With this spec we are trying to update these details as part of sync. So, no change to brick data model